src/helpers/unmountApplication.ts

```diff
@@ -0,0 +1,27 @@
+import { reconciler } from '../core/reconciler';
+import { roots } from '../core/roots';
+import { type Root } from '../typedefs/Root';
+
+export function unmountApplication(root: Root)
+{
+    if (root.internalState.queuedForUnmount)
+    {
+        const fiber = root?.fiber;
+
```

Comment:
If `root.internalState.queuedForUnmount` exists, then `root` exists, so uncertain why optional chain is used on line 9

Reply:
It's an artifact of a previous arch. Good catch.